src/monorepo-shipit/src/RepoGit.js

```diff
@@ -227,7 +229,17 @@ export default class RepoGit implements AnyRepo, SourceRepo, DestinationRepo {
       .runSynchronously()
       .getStdout();
     const trimmedLog = log.trim();
-    return trimmedLog === '' ? null : trimmedLog.split('\n');
+    if (trimmedLog === '') {
+      return null;
+    }
+
+    const descendants = trimmedLog.split('\n');
+
+    if (includeBaseRevision === true) {
+      descendants.unshift(baseRevision);
```

Comment:
Since the above command doesn't include the base revision we conditionally add it at the beginning here.

Reply:
I don't understand this change. Can you please explain it better?

I've used Shipit to export into new repositories many times, and it always worked well. Maybe the problem is somewhere else? 🤔

Reply:
So my initial commit on my monorepo isn't empty. Without this code this function never returns the first commit, but only the second. This means when changes are applied it starts with the second commit, and the diff is applying (via git am) to files that don't yet exist (because the git am application from the first commit hasn't happened!).

To get around this, without having to re-write my monorepos history to add an empty commit at the beginning of time, we can conditionally include the first commit (baseRevision) in the descendants and get around the problem.

Does that make sense?

Reply:
To be specific: initial commits are skipped by the `[findDescendantsPath](https://github.com/adeira/shipit/blob/master/src/RepoGit.js#L220-L223)` git command. This is because git log --ancestory-path A..B doesn't include the "A" commit. This is a problem when the initial commit has content like my repo does.

An alternative could be to instead get every commit if it's fresh destination repo? Both would be equivalent.

Reply:
If you're interested we could catch up on Discord and I can walk you through my monorepo and needs I have from this and beyond? 🙏

Reply:
@itsdouges Would you please share with me the exact steps to reproduce? What did you do exactly? I am still a bit lost and I would like to try it myself.

Shipit starts exporting from the beginning of the monorepo history if the child repo is either empty or doesn't contain the `adeira-source-id` handle. I've done both many times (export to empty child repo as well as connect Shipit to already existing child repo) and I know it works. Perhaps I just need to explain the process better?

Reply:
But that's the thing it doesn't do it from the first commit but the second. Let me find a way to reproduce in a unit test and I'll push it up.

Reply:
I pushed a test. Does that explain things?

Reply:
I appreciate your patience and especially the test case - that not only helped me to understand the issue but it also guarantees I won't break this by accident in the future. Thanks! All clear now!